pkg/deploy/kubernetes_image_puller.go

```diff
@@ -42,19 +42,19 @@ type ImageAndName struct {
 
 // Reconcile the imagePuller section of the CheCluster CR.  If imagePuller.enable is set to true, install the Kubernetes Image Puller operator and create
 // a KubernetesImagePuller CR.  Add a finalizer to the CheCluster CR.  If false, remove the KubernetesImagePuller CR, uninstall the operator, and remove the finalizer.
-func ReconcileImagePuller(ctx *DeployContext) (reconcile.Result, error) {
+func ReconcileImagePuller(ctx *DeployContext) (*reconcile.Result, error) {
```

Comment:
Why do we have to change to pointer for `reconcile.Result`? In main controller [object is passed directly](https://github.com/eclipse-che/che-operator/blob/main/controllers/che/checluster_controller.go#L227). Moreover it will cause a lot of edit below.